src/nni_manager/core/nnimanager.ts

```diff
@@ -479,8 +479,6 @@ class NNIManager implements Manager {
                     };
                     const trialJobDetail: TrialJobDetail = await this.trainingService.submitTrialJob(trialJobAppForm);
                     this.trialJobsMaintainer.setTrialJob(trialJobDetail.id, Object.assign({}, trialJobDetail));
-                    // TO DO: to uncomment
-                    assert(trialJobDetail.status === 'WAITING');
```

Comment:
why removing this assert? In which case does trainingservice return non 'WAITING' status?

Reply:
When the submitted job is happened to be scheduled immediately after submit,  the status of trialJobDetail is not waiting,  now change to use a snapshot jobdetail object created by job maintainer.